pages/services/nifi/0.5.0-1.9.2/troubleshooting/diagnostictools/index.md

```diff
@@ -0,0 +1,153 @@
+---
+layout: layout.pug
+navigationTitle: Diagnostic Tools
+title: Diagnostic Tools
+menuWeight: 102
+excerpt: Troubleshooting clusters
+featureMaturity:
+enterprise: false
+---
+
+# Diagnostic Tools
+
+DC/OS clusters provide several tools for diagnosing problems with services running in the cluster. In addition, the SDK has its own endpoints that describe what the Scheduler is doing at any given time.
+
+## Logging
+
+The first step to diagnosing a problem is typically to take a look at the logs. Knowledge of the problem being diagnosed will help you to determine which task logs are relevant.
+
+As of this writing, the best and fastest way to view and download logs is via the Mesos UI at `<dcos-url>/mesos`. On the Mesos front page you will see two lists: A list of currently running tasks, followed by a list of completed tasks (whether successful or failed).
+
+![mesos frontpage showing all tasks in the cluster](../../img/1_Logging_All_Tasks.png)
+
+Figure 1. - Mesos front page
+
+The Sandbox link for one of these tasks shows a list of files from within the task itself. For example, here’s a sandbox view of a nifi-0-node task from the above list:
+
+![inside task" width](../../img/2_Inside_Task.png)
+
+Figure 2. - Task file list
+
+If the task is based on a Docker image, this list will only show the contents of `/mnt/sandbox`, and not the rest of the filesystem. If you need to view filesystem contents outside of this directory, you will need to use `dcos task exec` or `nsenter` as described below under Running commands within containers.
+
+In the above task list(figure 1) there are multiple services installed, resulting in a pretty large list. The list can be filtered using the text box at the upper right, but there may be duplicate names across services. For example there are two instances of nifi and they’re each running a node-0. As the cluster grows, this confusion gets proportionally worse. We want to limit the task list to only the tasks that are relevant to the service being diagnosed. To do this, click “Frameworks” on the upper left to see a list of all the installed frameworks (mapping to our services):
+
+![active frameworks](../../img/3_Active_Frameworks.png)
+
+Figure 3. - Active Frameworks
+
+We then need to decide which framework to select from this list. This depends on what task we want to view.
+
+## Scheduler Logs
+
+If the issue is one of deployment or management, for example, a service is ‘stuck’ in initial deployment, or a task that previously went down is not being brought back at all, then the Scheduler logs will likely be the place to find out why.
```

Comment:
This is a complicated structure for this sentence. I would rewrite it to break it up.